src/mds/Server.cc

```diff
@@ -9963,15 +9939,14 @@ void Server::handle_peer_rename_prep(MDRequestRef& mdr)
   mdr->ls = mdlog->get_current_segment();
   EPeerUpdate *le = new EPeerUpdate(mdlog, "peer_rename_prep", mdr->reqid, mdr->peer_to_mds,
 				      EPeerUpdate::OP_PREPARE, EPeerUpdate::RENAME);
-  mdlog->start_entry(le);
   le->rollback = mdr->more()->rollback_bl;
   
   bufferlist blah;  // inode import data... obviously not used if we're the peer
   _rename_prepare(mdr, &le->commit, &blah, srcdn, destdn, mdr->peer_request->alternate_name, straydn);
 
   if (le->commit.empty()) {
     dout(10) << " empty metablob, skipping journal" << dendl;
-    mdlog->cancel_entry(le);
+    delete mdr->ls;
```

Comment:
I think this is causing the MDS crash: https://pulpito.ceph.com/vshankar-2023-03-02_09:21:58-fs-wip-vshankar-testing-20230222.044949-testing-default-smithi/7191766/

```
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/18.0.0-2462-g802de851/rpm/el8/BUILD/ceph-18.0.0-2462-g802de851/src/include/elist.h: 91: FAILED ceph_assert(_head.empty())

 ceph version 18.0.0-2462-g802de851 (802de851643c97960b33dacc1e3ce8486e6d7d3c) reef (dev)                                                                                                                          1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x135) [0x56eedbb]
 2: /usr/lib64/ceph/libceph-common.so.2(+0x2a7f81) [0x56eef81]
 3: ceph-mds(+0x2c2263) [0x3ca263]                                                                                                                                                                                 4: (Server::handle_peer_rename_prep(boost::intrusive_ptr<MDRequestImpl>&)+0x1ccf) [0x3ad5df]                                                                                                                      5: (MDSContext::complete(int)+0x5f) [0x64642f]
 6: (MDSRank::_advance_queues()+0xaa) [0x2cb13a]
 7: (MDSRank::_dispatch(boost::intrusive_ptr<Message const> const&, bool)+0x1ce) [0x2cbc2e]                                                                                                                        8: (MDSRankDispatcher::ms_dispatch(boost::intrusive_ptr<Message const> const&)+0x5c) [0x2cc61c]                                                                                                                   9: (MDSDaemon::ms_dispatch2(boost::intrusive_ptr<Message> const&)+0x1bf) [0x2b695f]
 10: (Messenger::ms_deliver_dispatch(boost::intrusive_ptr<Message> const&)+0x478) [0x5964398]
 11: (DispatchQueue::entry()+0x50f) [0x596153f]                                                                                                                                                                    12: (DispatchQueue::DispatchThread::entry()+0x11) [0x5a26281]                                                                                                                                                     13: /lib64/libpthread.so.0(+0x81cf) [0x6e341cf]                                                                                                                                                                   14: clone()
```

Reply:
Cursory looks like it should be `delete mdr->le`. Will check tomorrow.